### ISSUE TITLE
Warnings

### ISSUE THREAD

**Issue by Carlosgrr** [Dav1dde/glad#206]
I have generated the glad library using the webservice with the options: Core and OpenGL 3.3, without any extensions. When trying to compile using GCC with -Wpedantic, it gives about 800 warnings, mostly:

warning: ISO C forbids conversion of object pointer to function pointer type [-Wpedantic]

**Comment by fluffrabbit** [comment]
Interesting. I use similar settings with Glad 2, -Wall -Wpedantic C++11 and get few or no warnings.

**Comment by Carlosgrr** [comment]
> Interesting. I use similar settings with Glad 2, -Wall -Wpedantic C++11 and get few or no warnings.
> […](#)
> On April 2, 2019 3:37:12 PM MDT, Carlosgrr ***@***.***> wrote: I have generated the glad library using the webservice with the options: Core and OpenGL 3.3, without any extensions. When trying to compile using GCC with -Wpedantic, it gives about 800 warnings, mostly: warning: ISO C forbids conversion of object pointer to function pointer type [-Wpedantic] -- You are receiving this because you are subscribed to this thread. Reply to this email directly or view it on GitHub: #206

Perhaps a differing version of GCC?
Mine is GCC 8.2.1, what's yours?

**Comment by fluffrabbit** [comment]
7.3.0. Huh. I'll have to test with 8.2.x when I get home.

**Comment by fluffrabbit** [comment]
Still compiles without warnings on my end with g++-8 (gcc 8.2.0) in C++11 mode with Glad 2 -Wall -Wpedantic.

**Comment by fluffrabbit** [comment]
I'm using the header-only version if that makes a difference.

**Comment by Carlosgrr** [comment]
I downloaded the full .zip package and included both the .c and the .h files, in fact the compiler errors are inside the .c file.

**Comment by Dav1dde** [comment]
Glad needs to convert object pointers to function pointers there is no way around it and this is against the standard, since this may fail on some platforms. But on any of those platforms OpenGL wouldn't or couldn't run either.

There are a few older issues for this and ideas to get around it and trick the compiler, I decided against tricking the compiler and doing something that would actually be undefined behavior.

Glad 2 on the other hand tells GCC it uses a C extension (object to function pointed) and makes GCC ignore the casts this way.

Tldr for glad 1 don't compile with pedantic. Or try glad 2.

Glad 2 can be found on https://glad.sh or the glad 2 branch in this repo. It is mainly finished but a few things still can change and I may break the API one last time.
